pkg/resolvers/k8s/streamer_test.go

```diff
@@ -137,103 +148,153 @@ func TestStreamWatcher_NotSupportedType_EventErr_ClosesConn(t *testing.T) {
 	require.NoError(t, err)
 
 	bytesCh <- []byte(b)
-	event := <-eventsCh
-	require.Error(t, event.err, "we expect it invalid event type error")
+	select {
+	case err := <-errCh:
+		require.Error(t, err, "we expect invalid event type error")
+		require.Equal(t, "got invalid watch event type: not-supported", err.Error())
+	case change := <-changeCh:
+		t.Errorf("unexpected change event %v", change)
+		t.FailNow()
+	}
 
 	// Expect connection closed.
 	select {
-	case <-eventsCh:
+	case <-errCh:
+		t.Error("No err was expected")
+	case <-changeCh:
 		t.Error("No event was expected")
-	// Not really nice to use time in tests, but should be enough for now.
+		// Not really nice to use time in tests, but should be enough for now.
+	case <-time.After(200 * time.Millisecond):
+	}
+	require.True(t, connMock.isClosed())
+}
+
+func TestStreamWatcher_ErrorEventType_EventErr_ClosesConn(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 10*time.Second)
+
+	bytesCh, _, connMock, streamer, cancel := startTestStream(t)
+	defer cancel()
+
+	changeCh, errCh := streamer.ResultChans()
+
+	status := &metav1.Status{
+		Status:  "err",
+		Code:    124,
+		Message: "some-msg",
+	}
+	// Triggering not supported event.
+	wrongEvent := event{
+		Type: watch.Error,
+		Object: eventObject{
+			Status: status,
+		},
+	}
+	b, err := json.Marshal(wrongEvent)
+	require.NoError(t, err)
+
+	bytesCh <- []byte(b)
+	select {
+	case err := <-errCh:
+		require.Error(t, err, "we expect event type error")
+		require.Equal(t, "err: some-msg. Code: 124", err.Error())
+	case change := <-changeCh:
+		t.Errorf("unexpected change event %v", change)
+		t.FailNow()
+	}
+
+	// Expect connection closed.
+	select {
+	case <-errCh:
+		t.Error("No err was expected")
+	case <-changeCh:
+		t.Error("No event was expected")
+		// Not really nice to use time in tests, but should be enough for now.
 	case <-time.After(200 * time.Millisecond):
 	}
 	require.True(t, connMock.isClosed())
 }
 
 func TestStreamWatcher_EOF_EventErr_ClosesConn(t *testing.T) {
-	_, errCh, connMock, eventsCh, cancel := startTestStream(t)
+	defer leaktest.CheckTimeout(t, 10*time.Second)
+
+	_, connErrCh, connMock, streamer, cancel := startTestStream(t)
 	defer cancel()
 
+	changeCh, errCh := streamer.ResultChans()
+
 	// Triggering EOF.
-	errCh <- io.EOF
-	event := <-eventsCh
-	require.Error(t, event.err, "we expect it invalid event type error")
+	connErrCh <- io.EOF
+	select {
+	case err := <-errCh:
+		require.Error(t, err, "we expect EOF error")
+		require.Equal(t, err, io.EOF)
+	case change := <-changeCh:
+		t.Errorf("unexpected change event %v", change)
+		t.FailNow()
+	}
 
 	// Expect connection closed.
 	select {
-	case <-eventsCh:
+	case <-errCh:
+		t.Error("No err was expected")
+	case <-changeCh:
 		t.Error("No event was expected")
-	// Not really nice to use time in tests, but should be enough for now.
+		// Not really nice to use time in tests, but should be enough for now.
```

Comment:
?

Reply:
Do you use golang 1.10? It seems like they've changed something in a new version.